### PR TITLE
Add API endpoint to retrieve psm options

### PR DESCRIPTION
**Pull request by kolakachi**
Added an end point for fetching psm options for Tesseract

Bug: [T353824](http://phabricator.wikimedia.org/T353824)

### CODE REVIEW

File: src/Controller/OcrController.php [hunk 1]
```diff
@@ -311,6 +311,28 @@ public function apiAvailableLangsAction(): JsonResponse {
 		] );
 	}
 
+	/**
+	 * Get a list of Psms available for use with Tesseract.
+	 *
+	 * @Route("/api/tesseract/available_psms", name="apiPsms", methods={"GET"})
+	 * @OA\Response(response=200, description="List of available psm values and labels, in JSON format.")
```
Comment (Parthiv-M) [inline src/Controller/OcrController.php:318]: Maybe we should add _for tesseract_ here?

**Review by Parthiv-M — COMMENTED** [review]
Work fine for me. I was wondering whether we should remove the _Automatic page segmentation, but no OSD, or OCR. (not implemented)_ altogether? Ping @samwilson

**Comment by samwilson** [comment]
> I was wondering whether we should remove the _Automatic page segmentation, but no OSD, or OCR. (not implemented)_ altogether?

Probably a good idea, given that it's not meant to be used. I don't think many people have noticed that "(not implemented)"! :-) 

Although if that's removed, I'd suggest moving the PSM loop into a central place, because now we've got it in `templates/_tesseract_options.html.twig` and `OcrController::apiAvailablePsms()`. Actually, centralising it is probably worth doing in this patch (e.g. pass the output of `apiAvailablePsms()` into `OcrController::$params` as is done for `available_line_ids`).

**Comment by samwilson** [comment]
The other bits can be done in a follow-up.

**Pull request merged**